web/src/engine/websites/MangaLib.ts

```diff
@@ -1,46 +1,125 @@
 ﻿import { Tags } from '../Tags';
 import icon from './MangaLib.webp';
-import { DecoratableMangaScraper } from '../providers/MangaPlugin';
+import { Chapter, DecoratableMangaScraper, Manga, type MangaPlugin, Page } from '../providers/MangaPlugin';
 import * as Common from './decorators/Common';
+import { FetchJSON } from '../platform/FetchProvider';
+import { Priority, TaskPool } from '../taskpool/TaskPool';
+import { RateLimit } from '../taskpool/RateLimit';
 
-function MangaExtractor(anchor: HTMLAnchorElement) {
-    return {
-        id: anchor.pathname,
-        title: anchor.querySelector('.media-card__title').textContent.trim()
-    };
-}
-
-const chapterScript = `
-    new Promise(resolve => {
-        const chapters = window.__DATA__.chapters.list.map(entry => {
-            return {
-                id: '/' + window.__DATA__.manga.slug + '/v' + entry.chapter_volume + '/c' + entry.chapter_number,
-                title: entry.chapter_number + (entry.chapter_name ? ' - ' + entry.chapter_name : '')
-            };
-        });
-        resolve(chapters);
-    });
-`;
-
-const pageScript = `
-    new Promise(resolve => {
-        resolve(window.__pg.map(page => window.__info.servers.main + window.__info.img.url + page.u));
-    });
-`;
-
-@Common.MangaCSS(/^{origin}\/[^/]+(\?section=info)?$/, 'div.media-name__body div.media-name__main')
-@Common.MangasMultiPageCSS('/manga-list?page={page}', 'div.media-card-wrap a.media-card', 1, 1, 0, MangaExtractor)
-@Common.ChaptersSinglePageJS(chapterScript, 500)
-@Common.PagesSinglePageJS(pageScript, 500)
-@Common.ImageAjax()
+type ImageServers = {
+    imageServers: ImageServer[]
+}
+
+type ImageServer = {
+    id: string,
+    url: string,
+    site_ids: number[]
+}
+
+type APIResult<T> = {
+    data: T,
+    meta: {
+        has_next_page: boolean,
+    }
+}
+
+type APIManga = {
+    id: number,
+    rus_name: string,
+    slug_url: string
+}
+
+type APIChapter = {
+    volume: string,
+    number: string,
+    name: string,
+    branches_count: number,
+    branches: {
+        id: number,
+        branch_id: number | null,
+        teams: {
+            name: string
+        }[]
+    }[]
+}
+
+type APIPages = {
+    pages: {
+        url: string
+    }[]
+}
+
+type ChapterID = {
+    branch_id: string,
+    number: string,
+    volume: string
+}
+
+@Common.ImageAjax(true)
 export default class extends DecoratableMangaScraper {
+    private readonly apiUrl = 'https://api.mangalib.me/api/';
+    private readonly siteId = 1;
+    private imageServer: ImageServer = undefined;
```

Comment:
May just use type string (url) since `ImageServer.id` and `ImageServer.site_ids` are not used in `FetchImages`.